docs/glossary.md

```diff
@@ -31,3 +31,19 @@
   - Defending units can only participate in this phase if `Submarines Defending May Submerge Or Retreat` property is set.
   - `Sub Retreat Before Battle` defines if this phase occurs before the First Strike Phase or at the end of the Battle Round.
 - **Submerge Subs Phase**: Another name for **Evasion Phase** and generally used when only submarines have `canEvade` and `Submersible Subs` is set.
+
+##Development Terms
+- **Branch**: A Branch is a copy of the main codebase.  It is used to keep changes separate until they are completed or in a stable state.  
+- **Checkout**: When you Checkout a branch, that is where commits are stored.  
```

Comment:
Checkout is ambiguous, a checkout can mean multlipe things and depends on the version control system. In SVN and TFS, a checkout is downloading a copy of a branch or a set of branches. Git is not trivially consistent in defining a checkout and what it does. Generally a checkout creates relates to a branch head pointer. If you checkout out a new branch, you are setting your current HEAD pointer to the HEAD of a new branch (a git work space maintains a single pointer that references a single commit. When you checkout commits, branches, you are moving this pointer).

So, a checkout of a branch moves the HEAD pointer to HEAD of that branch. A checkout of a new branch creates a new branch from the current one and moves the HEAD pointer to the HEAD of that branch. A checkout of a file changes the contents of the file such that the contents matches the current HEAD pointer of the staged changes. 

Changes in git can be staged or unstaged. The unstaged changes means the working HEAD pointer is pointed at a commit, on a branch or otherwise. Staged changes are files you have added which creates a 'phantom' commit and your HEAD pointer refers to it. So if you `git add <file>`, it'll be staged and a checkout of that file will revert the contents back to what it was when it was staged, and any diff of contents will be compared to the staged contents.